### PR TITLE
chore: add a common execution id into CI events

### PR DESCRIPTION
**Pull request by ThomasRooney**

### CODE REVIEW

File: .github/workflows/build-docker.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 env:
   REGISTRY: ghcr.io
   IMAGE_NAME: ${{ github.repository }}
-
+  GOPRIVATE: "github.com/speakeasy-api"
```
Comment (TristanSpeakEasy) [inline .github/workflows/build-docker.yaml:11]: So this represents a significant change.

Before this our whole action is opensource, can be forked and modified as need but this makes it not possible without a big rewrite for someone

Comment (TristanSpeakEasy) [reply .github/workflows/build-docker.yaml:11]: can we do this without pulling in a private dependency

Comment (ThomasRooney) [reply .github/workflows/build-docker.yaml:11]: done.

File: internal/environment/environment.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ type Mode string
 
 const (
 	ModeDirect Mode = "direct"
-	ModePR     Mode = "pr"
+	ModePR                         Mode = "pr"
```
Comment (TristanSpeakEasy) [inline internal/environment/environment.go:15]: formatting?

**Pull request merged**